src/plugins/intel_gpu/src/runtime/device.cpp

```diff
@@ -4,64 +4,128 @@
 
 #include "intel_gpu/runtime/device.hpp"
 #include "intel_gpu/runtime/debug_configuration.hpp"
-
+#include <cmath>
+#include <limits>
 namespace cldnn {
 
-float device::get_gops(cldnn::data_types dt) const {
-    auto info = get_info();
-    if (info.vendor_id != INTEL_VENDOR_ID) {
-        // GOPS calculation is not supported for non Intel GPUs
-        return 0.0f;
-    }
-    auto freqGHz = info.gpu_frequency / 1000.f;
-    auto numEUs = info.execution_units_count;
-    auto opsPerComputeBlock = 0;
-    auto computeBlockIPC = 1.0f;
-    switch (dt) {
-    case cldnn::data_types::u8:
-    case cldnn::data_types::i8: {
-        if (info.supports_immad) {
-            if (info.gfx_ver.major == 12) {
-                if (info.gfx_ver.minor == 5)
-                    opsPerComputeBlock = 512;
-                else if (info.gfx_ver.minor == 7)
-                    opsPerComputeBlock = 256;
+
+struct DeviceOps {
+    DeviceOps(std::string platform,
+              std::vector<uint32_t> dev_id_list,
+              std::vector<gfx_version> gfx_version_list,
+              std::vector<float> ops): platform(platform), dev_id_list(dev_id_list), gfx_version_list(gfx_version_list) {
+        auto add_ops_to_map = [](std::map<data_types, float>& out_map, std::vector<float> ops, data_types dt, size_t index) -> void {
+            if (!is_zero(ops[index])) {
+                out_map[dt] = ops[index];
             }
-        } else if (info.supports_imad) {
-            // fma * simd size
-            opsPerComputeBlock = 2 * 32;
-        } else {
-            // separate mul + add instructions for int8 data type
-            opsPerComputeBlock = 2 * 16;
-            // mul/add instructions can't be executed in parallel, so we need 2 clocks to execute compute block
-            computeBlockIPC = 0.5f;
+        };
+
+        add_ops_to_map(ops_mad,  ops, data_types::f64, 0);
+        add_ops_to_map(ops_mad,  ops, data_types::f32, 1);
+        add_ops_to_map(ops_mad,  ops, data_types::f16, 2);
+        add_ops_to_map(ops_dpas, ops, data_types::f16, 3);
+        add_ops_to_map(ops_dpas, ops, data_types::i8,  4);
+        add_ops_to_map(ops_dpas, ops, data_types::i4,  5);
+        add_ops_to_map(ops_dp4a, ops, data_types::f16, 6);
+        add_ops_to_map(ops_dp4a, ops, data_types::i8,  7);
+    }
+
+    std::string platform;
+    std::vector<uint32_t> dev_id_list;
+    std::vector<gfx_version> gfx_version_list;
+    std::map<data_types, float> ops_mad;
+    std::map<data_types, float> ops_dpas;
+    std::map<data_types, float> ops_dp4a;
+
+    static bool is_zero(float value) {
+        return std::abs(value) <= std::numeric_limits<float>::epsilon();
+    }
+
+    bool is_support(device_info& info) const {
```

Comment:
nit: misleading name: `is_support` -> `match`

Reply:
apply it